File: presets/www/dcurgz.me/templates/fibonacci.hs
-- fibonacci.hs: A Haskell implementation of the Fibonacci function.
import System.Environment (getArgs)
import System.Exit
import Data.Function ((&))

fib :: Integer -> Integer
fib 0 = 0
fib 1 = 1
fib n = fib (n-1) + fib (n-2)

main :: IO ()
main = do
    args <- getArgs
    if null args
        then do
            putStrLn "A fibonacci implementation written in Haskell.\n"
            putStrLn "Compute fibonacci(n) for any given n.\n"
            putStrLn "Usage: fibonacci <number>"
            exitSuccess
        else
            args & head & read & fib & putStr . show

File: nix/modules/www/dcurgz.me/webroot/pages/nixos/_drafts/fibonacci.c
// fibonacci.c: A C implementation of the Fibonacci function.
#include <stdio.h>
#include <stdlib.h>

int fib(int n) {
    if (n < 2)
        return n;
    int i1 = 1, i2 = 0, tmp = 0;
    for (int i = 2; i < n; i++) {
        tmp = i1;
        i1 = i1 + i2;
        i2 = tmp;
    }
    return i1 + i2;
}

int main(int argc, char **argv) {
    if (argc < 2) {
        printf("A fibonacci implementation in C.\\n");
        printf("Computes fibonacci(n) for any given n.\\n");
        printf("Usage: %s <number>", argv[0]);
        exit(0);
    }
    int n = atoi(argv[1]);
    int result = fib(n);
    printf("%d", result);
}
